qolmat/analysis/holes_characterization.py

```diff
@@ -1,18 +1,32 @@
 from abc import ABC, abstractmethod
-from typing import Optional, Union
+from itertools import combinations
+from typing import List, Optional, Tuple, Union
 
+from category_encoders.one_hot import OneHotEncoder
+from joblib import Parallel, delayed
 import numpy as np
 import pandas as pd
+from sklearn.ensemble import RandomForestClassifier
+from sklearn import utils as sku
 from scipy.stats import chi2
 
 from qolmat.imputations.imputers import ImputerEM
+from qolmat.utils.input_check import check_pd_df_dtypes
 
 
 class McarTest(ABC):
     """
     Astract class for MCAR tests.
+
+    Parameters
+    ----------
+    random_state : int, optional
+        The seed of the pseudo random number generator to use, for reproductibility.
     """
 
+    def __init__(self, random_state: Union[None, int, np.random.RandomState] = None):
+        self.rng = sku.check_random_state(random_state)
+
     @abstractmethod
     def test(self, df: pd.DataFrame) -> float:
```

Comment:
add docstring ? 
+ modify the return type, because sometime, it is a Union[float, Tuple[float, List[float]]] as in the PKLMTest class.